core/templating/template_helpers.go

```diff
@@ -398,6 +496,56 @@ func (t templateHelpers) csvCountRows(dataSourceName string) string {
 	return fmt.Sprintf("%d", numRows)
 }
 
+func (t templateHelpers) csvSQL(queryString string) []RowMap {
+	templateDataSources := t.TemplateDataSource.DataSources
+	// Parse the query string to get the SelectQuery
+	query, err := parseQuery(strings.TrimSpace(queryString), templateDataSources)
+	if err != nil {
+		// Debugging: Print the parsed query
+		fmt.Printf("Error parsing query:: %+v\n", err)
+		log.Debug("Error parsing query:", err)
+		return []RowMap{}
+	}
+
+	// Find the data source by name
+	source, exists := templateDataSources[query.DataSourceName]
+	if !exists {
+		log.Debug("Could not find datasource " + query.DataSourceName)
+		return []RowMap{}
+	}
+
+	source.mu.Lock()
+	defer source.mu.Unlock()
+
+	var results []RowMap
+	var execErr error
+
+	switch query.Type {
+	case "SELECT":
+		results = executeSelectQuery(source.Data, query) //This is not by reference as we are not changing the data
+	case "UPDATE":
+		execErr = executeUpdateQuery(&source.Data, query) //This must be by reference as we are not changing the data
+	case "DELETE":
+		execErr = executeDeleteQuery(&source.Data, query) //This must be by reference as we are not changing the data
```

Comment:
sql update/ delete returns number of affected rows and perhaps we should replicate it, and log a message as well so that we know something has happened

Reply:
Yes, I will change it

Reply:
Done

Reply:
I have updated update and delete to log rows and return nil